### PR TITLE
Remove print usage in favor of logging

### PR DESCRIPTION
**Pull request by pokoli**
As commented in #40 it will be great to remove all the print usage and stick to configurable logging.

### CODE REVIEW

**Comment by mar10** [comment]
I cleaned up the whitespace.
Could you update your branch from master, so we hopefully reduce your PR to the relevant changes?

**Comment by pokoli** [comment]
@mar10 I rebased my PR on top of the master branch. It looks nicer now and I think it will be easier to review now.

File: wsgidav/lock_manager.py [hunk 1]
```diff
@@ -150,7 +148,7 @@ def _dump(self, msg="", out=None):
         userDict = {} # { <LOCKUSER>: [<tokenlist>] }
         tokenDict = {} # { <token>: <LOCKURLS> }
 
-        print("%s: %s" % (self, msg), file=out)
```
Comment (mar10) [inline wsgidav/lock_manager.py]: Since the 'out' argument is no longer used, we should remove it

File: wsgidav/wsgidav_app.py [hunk 2]
```diff
@@ -213,23 +212,20 @@ def __init__(self, config):
...
+        _logger.debug("Using property manager: %r" % propsManager)
+        _logger.debug("Using domain controller: %s" % domain_controller)
+        _logger.debug("Registered DAV providers:")
+        for share, data in self.providerMap.items():
+            hint = " (anonymous)" if data['allow_anonymous'] else ""
+            _logger.debug("  Share '%s': %s%s" % (share, provider, hint))
+        for share, data in self.providerMap.items():
+            if data['allow_anonymous']:
+                # TODO: we should only warn here, if --no-auth is not given
+                _logger.warning("WARNING: share '%s' will allow anonymous access." % share)
```
Comment (mar10) [inline wsgidav/wsgidav_app.py]: 'anonymous access' warning was issued on verbose >= 1, i. e. level INFO

File: wsgidav/lock_manager.py (hunk 1 above)
Comment (mar10) [inline wsgidav/lock_manager.py]: Same for property_manager.py?

**Comment by mar10** [comment]
Thanks. (I think we wait with resolving and merging until the majority of the pep8 changes are applied?)

**Comment by djmattyg007** [comment]
It'd be great if this could get merged. I'm experiencing some 502 errors when attempting to rename files and I'm struggling to work out why.

**Comment by pokoli** [comment]
Sorry but I'm not able to work on this on the next days. So feel free to pick up the PR and update them to tip.

**Comment by mar10** [comment]
I finally managed to start this refactoring. Your PR had too many conflicts to merge, and also I addressed some more PEP8 and %s => format() issues.
However, I mentioned you as co-comitter ;-)

**Pull request closed without merging**